### PR TITLE
refactor: update `blas/ext/base/dapxsumkbn` to follow current project conventions

### LINKED ISSUE #1466: [RFC]: refactor `blas/ext/base/dapxsumkbn` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dapxsumkbn`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dapxsumkbn) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by kumarsuraj212003**
Resolves #1466 .

## Description

> What is the purpose of this pull request?

This pull request: refactor `blas/ext/base/dapxsumkbn` to follow current project conventions.

## Related Issues 
Related issues #788, #1152.

> Does this pull request have any related issues?

This pull request:

-   resolves #1466
-   progresses #1152

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.js [hunk 1]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float64Array = require( '@stdlib/array/float64' );
 var pkg = require( './../package.json' ).name;
 var dapxsumkbn = require( './../lib/dapxsumkbn.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.native.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.native.js [hunk 4]
```diff
@@ -36,6 +36,7 @@ var dapxsumkbn = tryRequire( resolve( __dirname, './../lib/dapxsumkbn.native.js'
 var opts = {
 	'skip': ( dapxsumkbn instanceof Error )
 };
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.ndarray.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.ndarray.js [hunk 6]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float64Array = require( '@stdlib/array/float64' );
 var pkg = require( './../package.json' ).name;
 var dapxsumkbn = require( './../lib/ndarray.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.ndarray.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.ndarray.native.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.ndarray.native.js [hunk 8]
```diff
@@ -36,6 +36,7 @@ var dapxsumkbn = tryRequire( resolve( __dirname, './../lib/ndarray.native.js' )
 var opts = {
 	'skip': ( dapxsumkbn instanceof Error )
 };
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/benchmark/benchmark.ndarray.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/examples/index.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/examples/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/examples/index.js [hunk 10]
```diff
@@ -18,18 +18,11 @@
...
 var dapxsumkbn = require( './../lib' );
 
-var x;
-var i;
-
-x = new Float64Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-	x[ i ] = round( randu()*100.0 );
-}
+var x = filledarrayBy( 10, 'float64', discreteUniform( -100.0, 100.0 ) );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/examples/index.js]:
```suggestion
var x = filledarrayBy( 10, 'float64', discreteUniform( 0.0, 100.0 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/lib/index.js [hunk 11]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/lib/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/lib/ndarray.js [hunk 12]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/lib/ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/lib/ndarray.native.js [hunk 13]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/lib/ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/lib/ndarray.native.js [hunk 14]
```diff
@@ -38,20 +39,16 @@ var addon = require( './dapxsumkbn.native.js' );
...
 * // returns 25.0
 */
 function dapxsumkbn( N, alpha, x, stride, offset ) {
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float64Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+	offset = minViewBufferIndex( N, stride, offset );
+	view = offsetView( x, offset );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/lib/ndarray.native.js]:
```suggestion

	offset = minViewBufferIndex( N, stride, offset );

	view = offsetView( x, offset );

```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js [hunk 15]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js [hunk 16]
```diff
@@ -110,7 +109,7 @@ tape( 'the function supports a `stride` parameter', function test( t ) {
 		2.0
 	]);
 
-	N = floor( x.length / 2 );
+	N = 4;
 	v = dapxsumkbn( N, 5.0, x, 2 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js]:
```suggestion
	v = dapxsumkbn( 4, 5.0, x, 2 );
```

Remove `N` declaration

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js [hunk 17]
```diff
@@ -133,7 +132,7 @@ tape( 'the function supports a negative `stride` parameter', function test( t )
 		2.0
 	]);
 
-	N = floor( x.length / 2 );
+	N = 4;
 	v = dapxsumkbn( N, 5.0, x, -2 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js]:
```suggestion
	v = dapxsumkbn( 4, 5.0, x, -2 );
```
same comment here

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js [hunk 18]
```diff
@@ -171,8 +170,7 @@ tape( 'the function supports view offsets', function test( t ) {
 	]);
 
 	x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
-	N = floor(x1.length / 2);
-
+	N = 4;
 	v = dapxsumkbn( N, 5.0, x1, 2 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js]: same here

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.native.js [hunk 19]
```diff
@@ -289,8 +288,7 @@ tape( 'the function supports view offsets', opts, function test( t ) {
 	]);
 
 	x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
-	N = floor(x1.length / 2);
-
+	N = 4;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.native.js]: same comments

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.native.js [hunk 20]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.ndarray.js [hunk 21]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.ndarray.js [hunk 22]
```diff
@@ -167,7 +166,7 @@ tape( 'the function supports an `offset` parameter', function test( t ) {
 		3.0,
 		4.0   // 3
 	]);
-	N = floor( x.length / 2 );
+	N = 4;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.ndarray.js]: same comments

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.ndarray.native.js [hunk 23]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.ndarray.native.js [hunk 24]
```diff
@@ -176,7 +175,7 @@ tape( 'the function supports an `offset` parameter', opts, function test( t ) {
 		3.0,
 		4.0   // 3
 	]);
-	N = floor( x.length / 2 );
+	N = 4;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.ndarray.native.js]: same here

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Changes requested. Thanks for your works!

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/manifest.json [hunk 25]
```diff
@@ -34,7 +34,12 @@
 				"-lm"
 			],
 			"libpath": [],
-			"dependencies": []
+			"dependencies": [
+				"stdlib/napi/export.h",
+				"stdlib/napi/argv.h",
+				"stdlib/napi/argv_int64.h",
+				"stdlib/napi/argv_strided_float64array.h"
+			]
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/manifest.json]: This whole file is TAB indented, can you please fix this?

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js [hunk 26]
```diff
@@ -171,9 +168,8 @@ tape( 'the function supports view offsets', function test( t ) {
 	]);
 
 	x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
-	N = floor(x1.length / 2);
-
-	v = dapxsumkbn( N, 5.0, x1, 2 );
+	N = 4;
+	v = dapxsumkbn( 4, 5.0, x1, 2 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js]: please remove variable N declaration

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.native.js (hunk 19 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.native.js]: @kumarsuraj212003 this is not done yet

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Left comments

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/manifest.json (hunk 25 above)
Comment (kumarsuraj212003) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/manifest.json]: Thanks for the review. I fixed everything you suggested.

**Comment by kumarsuraj212003** [comment]
pinging @Pranavchiku for the review

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/src/addon.c [hunk 27]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+* Receives JavaScript callback invocation data.
+*
+* @private
+* @param env    environment under which the function is invoked
+* @param info   callback data
+* @return       Node-API value
+*/
+static napi_value addon(napi_env env, napi_callback_info info) {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/src/addon.c]:
```suggestion
static napi_value addon( napi_env env, napi_callback_info info ) {
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/src/addon.c [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+    STDLIB_NAPI_ARGV(env, info, argv, argc, 4);
+    STDLIB_NAPI_ARGV_INT64(env, N, argv, 0);
+    STDLIB_NAPI_ARGV_DOUBLE(env, alpha, argv, 1);
+    STDLIB_NAPI_ARGV_INT64(env, stride, argv, 3);
+    STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY(env, X, N, stride, argv, 2);
+
+    napi_value v;
+    napi_status status = napi_create_double(env, stdlib_strided_dapxsumkbn(N, alpha, (double *)X, stride), &v);
+    assert(status == napi_ok);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/src/addon.c]:
```suggestion
    STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
    STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
    STDLIB_NAPI_ARGV_DOUBLE( env, alpha, argv, 1 );
    STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
    STDLIB_NAPI_ARGV_STRIDED_FLOAT64ARRAY( env, X, N, stride, argv, 2 );

    napi_value v;
    napi_status status = napi_create_double( env, stdlib_strided_dapxsumkbn(N, alpha, X, stride), &v );
    assert( status == napi_ok );
```

**Review by Planeshifter — APPROVED** [review]
Thank you! Made a few edits, but this should now be ready to be merged. Will do so after CI has passed.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js [hunk 29]
```diff
@@ -171,9 +165,8 @@ tape( 'the function supports view offsets', function test( t ) {
 	]);
 
 	x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
-	N = floor(x1.length / 2);
-
-	v = dapxsumkbn( N, 5.0, x1, 2 );
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.js]:
```suggestion

```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.native.js [hunk 30]
```diff
@@ -289,9 +283,8 @@ tape( 'the function supports view offsets', opts, function test( t ) {
 	]);
 
 	x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
-	N = floor(x1.length / 2);
-
-	v = dapxsumkbn( N, 5.0, x1, 2 );
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn/test/test.dapxsumkbn.native.js]:
```suggestion

```

**Pull request merged**